words2num/lang_ES_US.py

```diff
@@ -0,0 +1,267 @@
+    'decillion': (10**33, 'X'),
+    'undecillion': (10**36, 'X'),
+    'duodillion': (10**39, 'X'),
+    'tredecillion': (10**42, 'X'),
+    'quattuordecillion': (10**45, 'X'),
+    'quindecillion': (10**48, 'X'),
+    'sexdecillion': (10**51, 'X'),
+    'septendecillion': (10**54, 'X'),
+    'octodecillón': (10**57, 'X'),
+    'novemdecillion': (10**60, 'X'),
+    'vigintillion': (10**63, 'X'),
+    'centillón': (10**303, 'X')
+}
+
+
+class FST:
+    def __init__(self):
+        def f_zero(self, n):
+            assert n == 0
+            self.value = n
+
+        def f_add(self, n):
+            self.value += n
+
+        def f_mul(self, n):
+            output = self.value * n
+            self.value = 0
+            return output
+
+        def f_mul_hundred(self, n):
+            assert n == 100
+            self.value *= n
+
+        def f_mul_hundred_and_add(self, n):
+            self.value *= 100
+            self.value += n
+
+        def f_ret(self, _):
+            return self.value
+
+        self.value = 0
+        self.state = 'S'
+        # self.states = {'S', 'D', 'T', 'M', 'H', 'X', 'Z', 'A', 'F'}
+        self.edges = {
+            ('S', 'Z'): f_zero,    # 0
+            ('S', 'D'): f_add,     # 9
+            ('S', 'T'): f_add,     # 90
+            ('S', 'M'): f_add,     # 19
+ #           ('S', 'A'): f_add,    # 100
+            ('S', 'H'): f_add,    # 100
+            ('S', 'F'): f_ret,     # 1
+#            ('D', 'H'): f_mul_hundred,     # 900
+            ('D', 'X'): f_mul,     # 9000
+            ('D', 'F'): f_ret,     # 9
+            ('T', 'D'): f_add,     # 99
+ #           ('D', 'T'): f_mul_hundred_and_add,     # 990 (nine ninety)
+ #          ('D', 'M'): f_mul_hundred_and_add,     # 919 (nine nineteen)
+ #           ('T', 'H'): f_mul_hundred,
+            ('T', 'X'): f_mul,     # 90000
+            ('T', 'F'): f_ret,     # 90
+ #           ('M', 'H'): f_mul_hundred,
+            ('M', 'X'): f_mul,     # 19000
+            ('M', 'F'): f_ret,     # 19
+            ('H', 'D'): f_add,     # 909
+            ('H', 'T'): f_add,     # 990
+            ('H', 'M'): f_add,     # 919
+            ('H', 'X'): f_mul,     # 900000
+            ('H', 'F'): f_ret,     # 900
+            ('X', 'D'): f_add,     # 9009
+            ('X', 'T'): f_add,     # 9090
+            ('X', 'M'): f_add,     # 9019
+            ('X', 'H'): f_add,     # 9900
+            ('X', 'F'): f_ret,     # 9000
+            ('Z', 'F'): f_ret,     # 0
+ #           ('A', 'H'): f_mul_hundred,     # 100
+ #           ('A', 'X'): f_mul,      # 1000
+            ('S', 'X'): f_add,      # 1000
+ #           ('A', 'F'): f_ret,      # 1
+        }
+
+    def transition(self, token):
+        value, label = token
+        try:
+            edge_fn = self.edges[(self.state, label)]
+        except KeyError:
+            raise NumberParseException("Invalid number state from "
+                                       "{0} to {1}".format(self.state, label))
```

Comment:
would be nice to use f-string formatting instead of "old" `.format()`
https://realpython.com/python-f-strings/